satpy/readers/viirs_edr.py

```diff
@@ -276,19 +277,28 @@ def _dynamic_variables_from_file(self, handled_var_names: set) -> Iterable[tuple
                 "resolution": res,
                 "coordinates": m_coords if res == 750 else i_coords,
             }
-            if is_lon:
-                ds_info["standard_name"] = "longitude"
-                ds_info["units"] = "degrees_east"
-                ds_info["name"] = m_lon_name if res == 750 else i_lon_name
-                # recursive coordinate/SwathDefinitions are not currently handled well in the base reader
-                del ds_info["coordinates"]
-            elif is_lat:
-                ds_info["standard_name"] = "latitude"
-                ds_info["units"] = "degrees_north"
-                ds_info["name"] = m_lat_name if res == 750 else i_lat_name
-                # recursive coordinate/SwathDefinitions are not currently handled well in the base reader
-                del ds_info["coordinates"]
-            yield True, ds_info
+            if not (is_lon or is_lat):
+                yield True, ds_info
+                continue
+
+            ds_info["standard_name"] = "longitude" if is_lon else "latitude"
+            ds_info["units"] = "degrees_east" if is_lon else "degrees_north"
+            # recursive coordinate/SwathDefinitions are not currently handled well in the base reader
+            del ds_info["coordinates"]
+            if var_name not in handled_var_names:
+                handled_var_names.add(var_name)
+                yield True, ds_info
+
+            # "standard" geolocation coordinate (assume shorter variable name is "better")
+            new_name = (m_lon_name if res == 750 else i_lon_name) if is_lon else (
+                m_lat_name if res == 750 else i_lat_name)
+            if new_name not in coords or len(var_name) < len(coords[new_name]["file_key"]):
```

Comment:
Would it be better to check explicitly for the parallax correction in the variable name? Or are there multiple of these that have `"longitude"` in the name? This name-length check would fail if they decided to add `lon` and `lat` as the "normal" coordinate arrays and the reader would need to adapted for them.

Reply:
Good question. You're right that the more explicit/specific solution would be to just check for the exact case I'm solving for. However, these parallax correction versions only show up (as far as I can find) in the "CloudHeight" files. Since this reader is supposed to be generic and potentially able to handle *any* VIIRS EDR file I'm worried about a hardcoded solution like that being not useful for these cases.

Side note: The SurfRefl files have multiple lon/lat arrays for the different resolutions of VIIRS, but I had to do some other customizations for these data files so they are hardcoded in the YAML.